plugins/trace_api_plugin/include/eosio/trace_api/compressed_file.hpp

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <ios>
+#include <fc/io/cfile.hpp>
+
+namespace eosio::trace_api {
+
+   class compressed_file_datastream;
+   struct compressed_file_impl;
+   /**
+    * wrapper for read-only access to a compressed file.
+    * compressed files support seeking and reading
+    *
+    * the efficiency of seeking is lower than that of an uncompressed file as each seek translates to
+    *  - 2 seeks + 1 read to load and process the seek-point-mapping
+    *  - potentially a read/decompress/discard of the data between the seek point and the requested offset
+    *
+    * More seek points can lower the average amount of data that must be read/decompressed/discarded in order
+    * to seek to any given offset.  However, each seek point has some effect on the file size as it represents a
+    * flush of the compressor which can degrade compression performance.
+    */
+   class compressed_file {
+   public:
+      explicit compressed_file( fc::path file_path );
+      ~compressed_file();
+
+      /**
+       * Provide default move construction/assignment
+       */
+      compressed_file( compressed_file&& );
+      compressed_file& operator= ( compressed_file&& );
+
+
+      void open() {
+         file_ptr = std::make_unique<fc::cfile>();
+         file_ptr->set_file_path(file_path);
+         file_ptr->open("rb");
+      }
+
+      bool is_open() const { return (bool)file_ptr; }
+
+      void seek( long loc );
```

Comment:
Precondition is that the file is not empty. Would be nice for this method to fail better so that `compressed_file` could be more general purpose.

Reply:
Actually I guess the `cfile::seek` will throw a `std::ios_base::failure` if unable to seek to location.